File: 0739-daily-temperatures/0739-daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (temperatures) {

    const lng = temperatures.length
    var out = new Array(lng).fill(0)
    var cache = []

    for (let i = 0; i < lng; i++) {
        while (cache.length > 0 && temperatures[i] > temperatures[cache[cache.length - 1]])
        {
            const val = cache.pop()
            out[val]= i - val
        }
        cache.push(i)
    }
    return out
};

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        lng = len(temperatures)
        out = [0]*lng
        cache = []

        for i in range(lng):
            while cache and temperatures[i] > temperatures[cache[-1]]:
                val = cache.pop()
                out[val] = i - val
            cache.append(i)
        return out